### PR TITLE
Fix message rustdocs

### LINKED ISSUE #774: messages rustdocs

### PR DESCRIPTION
**Pull request by raphaelrobert**
Fixes #774.

Note that this PR reshuffles some code for types according to their visibility, but the code is not changed.

### CODE REVIEW

File: openmls/src/messages/mod.rs [hunk 1]
```diff
@@ -146,6 +105,39 @@ impl Welcome {
     }
 }
 
+/// EncryptedGroupSecrets
+///
+/// This is part of a [`Welcome`] message. It can be used to correlate the correct secrets with each new member.
+#[derive(Clone, Debug, PartialEq, TlsDeserialize, TlsSerialize, TlsSize)]
+pub struct EncryptedGroupSecrets {
+    /// Key package refrence of the new member
```
Comment (kkohbrok) [inline openmls/src/messages/mod.rs:113]:
```suggestion
    /// Key package reference of the new member
```

File: openmls/src/messages/mod.rs [hunk 2]
```diff
@@ -146,6 +105,39 @@ impl Welcome {
...
+impl EncryptedGroupSecrets {
+    /// Build a new [`EncryptedGroupSecrets`].
+    pub fn new(new_member: KeyPackageRef, encrypted_group_secrets: HpkeCiphertext) -> Self {
+        Self {
+            new_member,
+            encrypted_group_secrets,
+        }
+    }
+
+    /// Returns the encrypted group secrets's new member [`KeyPackageRef`].
```
Comment (kkohbrok) [inline openmls/src/messages/mod.rs:128]:
```suggestion
    /// Returns the encrypted group secrets' new [`KeyPackageRef`].
```

File: openmls/src/messages/mod.rs [hunk 3]
```diff
@@ -146,6 +105,39 @@ impl Welcome {
...
+    pub fn new_member(&self) -> KeyPackageRef {
+        self.new_member
+    }
+
+    /// Returns a reference to the encrypted group secrets's encrypted group secrets.
```
Comment (kkohbrok) [inline openmls/src/messages/mod.rs:133]:
```suggestion
    /// Returns a reference to the encrypted group secrets' encrypted group secrets.
```

File: openmls/src/messages/proposals.rs [hunk 4]
```diff
@@ -339,17 +278,76 @@ impl GroupContextExtensionProposal {
...
+///   (255)
+/// } ProposalOrRefType;
+///
+/// struct {
+///   ProposalOrRefType type;
+///   select (ProposalOrRef.type) {
+///     case proposal:  Proposal proposal;
+///     case reference: opaque hash<0..255>;
+///   }
+/// } ProposalOrRef;
```
Comment (kkohbrok) [inline openmls/src/messages/proposals.rs:298]: Maybe remove these excerpts from the spec? You seem to have removed them everywhere else and I don't think they're very helpful to the reader anyway.

File: openmls/src/messages/proposals.rs [hunk 5]
```diff
@@ -339,17 +278,76 @@ impl GroupContextExtensionProposal {
...
+#[derive(
+    PartialEq, Clone, Copy, Debug, TlsSerialize, TlsDeserialize, TlsSize, Serialize, Deserialize,
+)]
+#[repr(u8)]
+pub(crate) enum ProposalOrRefType {
+    Proposal = 1,
+    Reference = 2,
+}
+
+/// Type of Proposal, either by value or by reference
```
Comment (kkohbrok) [inline openmls/src/messages/proposals.rs:312]:
```suggestion
/// Type of Proposal, either by value or by reference.
```

**Review by kkohbrok — CHANGES_REQUESTED** [review]
Thanks for the PR! Just a few typos and one other nit.

File: openmls/src/messages/proposals.rs (hunk 4 above)
Comment (raphaelrobert) [reply openmls/src/messages/proposals.rs:298]: I only touched the public docs. I can do that as a follow-up and remove it everywhere in the codebase if you think that makes sense.

Comment (kkohbrok) [reply openmls/src/messages/proposals.rs:298]: No worries. We can remove them one by one when we actually touch the corresponding code.

**Review by kkohbrok — APPROVED** [review]
Thanks!

**Pull request merged**